### PR TITLE
[TRAFODION-2685] int type doesnot check boundary

### PR DESCRIPTION
**Pull request by mashengchen**

### CODE REVIEW

**Comment by mashengchen** [comment]
jenkins, retest

**Comment by mashengchen** [comment]
jenkins, retest

File: core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4ResultSet.java [hunk 1]
```diff
@@ -1372,7 +1372,7 @@ public long getLong(int columnIndex) throws SQLException {
 						}
 					} else {
 						throw TrafT4Messages.createSQLException(connection_.props_, connection_.getLocale(),
-								"numeric_out_of_range", null);
+								"numeric_out_of_range", bd.longValue());
```
Comment (selvaganesang) [inline core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4ResultSet.java]: Will the longValue convert the bd into  a proper long value? Should we display the original bd value instead?

Comment (mashengchen) [inline core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4ResultSet.java]: agree ,i will change this

**Comment by mashengchen** [comment]
they are the same result after the process, even not use toString

**Comment by DaveBirdsall** [comment]
Is this change ready to merge?

**Comment by selvaganesang** [comment]
On second look, there seems to be lot of unneeded code/confusion in the surround code. The error message that is supposed to contain the original data is not consistently done in all the error conditions. This change seems to take care a certain error message paths alone. 

So, I will let the author to decide if the code change needs to go in or out.  I am ok with either of the conclusions.

**Comment by mashengchen** [comment]
this change is try to fix int type not check boundary which given by client, so the emphasis is in TrafT4PreparedStatement.java, check Integer Boundary when type is INTEGER (sign), and check Unsigned Integer Boundary when type is INTEGER UNSIGNED.
and The error message is try to let the user know what's the value they given in the request, this will help user quickly locate the issues

**Pull request merged**